src/test/java/base/BaseTest.java

```diff
@@ -1,51 +1,35 @@
 package base;
 
-import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.chrome.ChromeDriver;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
 import pages.CatalogPage;
 import pages.ElectronicsPage;
 import pages.HomePage;
 import pages.SmartphonePage;
-
-import java.util.concurrent.TimeUnit;
+import webdriver.DriverFactory;
+import webdriver.DriverManager;
 
 public class BaseTest {
-    private WebDriver driver;
     protected HomePage homePage;
     protected CatalogPage catalogPage;
     protected ElectronicsPage electronicsPage;
     protected SmartphonePage smartphonePage;
 
-    public static final String ONLINER_URL = "https://www.onliner.by/";
-    public static final int IMPLICITLY_WAIT = 10;
-
     @BeforeClass
     public void setUpDriver() {
-        System.setProperty("webdriver.chrome.driver", "src/main/resources/chromedriver.exe");
-        driver = new ChromeDriver();
-        driver.manage().window().maximize();
-        driver.manage().timeouts().implicitlyWait(IMPLICITLY_WAIT, TimeUnit.SECONDS);
-        driver.get(ONLINER_URL);
+        DriverManager.setDriver(DriverFactory.getDriver(DriverFactory.BrowserType.CHROME));
     }
 
     @BeforeClass
     public void setUpPages() {
-        homePage = new HomePage(getDriver());
-        catalogPage = new CatalogPage(getDriver());
-        electronicsPage = new ElectronicsPage(getDriver());
-        smartphonePage = new SmartphonePage(getDriver());
+        homePage = new HomePage(DriverManager.getDriver());
```

Comment:
Why do you pass driver as parameter to the page constructor if you can access it anywhere you need? That's what we discussed